File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> inputArrayMap = new HashMap<>();
        for (int i=0; i < nums.length; i++) {
            int num = nums[i];
            int diff = target - num;

            if (inputArrayMap.containsKey(diff)) {
                return new int[] {inputArrayMap.get(diff), i};
            }

            inputArrayMap.put(num, i);
        }

        return new int[] {};
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-0.kt
class Solution {
    fun hasDuplicate(nums: IntArray): Boolean {
        val hasDuplicate = HashSet<Int>()

        for (num in nums) {
            if (num in hasDuplicate) {
                return true
            }
            hasDuplicate.add(num)
        }
        return false
    }
}
